src/main/java/net/glowstone/command/minecraft/FillCommand.java

```diff
@@ -0,0 +1,249 @@
+        double zmin = 0;
+        double xmax = 0;
+        double ymax = 0;
+        double zmax = 0;
+        double placeX;
+        double placeY;
+        double placeZ;
+        if (x1 > x2) {
+            xmin = x2;
+            xmax = x1;
+        } else {
+            xmax = x2;
+            xmin = x1;
+        }
+        if (y1 > y2) {
+            ymin = y2;
+            ymax = y1;
+        } else {
+            ymax = y2;
+            ymin = y1;
+        }
+        if (z1 > z2) {
+            zmin = z2;
+            zmax = z1;
+        } else {
+            zmax = z2;
+            zmin = z1;
+        }
+        double blockAmount = (xmax - xmin) * (ymax - ymin) * (zmax - zmin);
+        if (blockAmount > 40000) {
+            new LocalizedStringImpl("fill.tooManyBlocks",
+            commandMessages.getResourceBundle())
+                .sendInColor(ChatColor.RED, sender, Double.toString(blockAmount));
+            return false;
+        }
+        int placementMode = 0;
+        if (args.length > 8) {
+            if (args[8].equalsIgnoreCase("outline")) {
+                placementMode = 1;
+            }
+            if (args[8].equalsIgnoreCase("hollow")) {
+                placementMode = 2;
+            }
+            if (args[8].equalsIgnoreCase("keep")) {
+                placementMode = 3;
+            }
+            if (args[8].equalsIgnoreCase("destroy")) {
+                placementMode = 4;
+            }
+            if (args[8].equalsIgnoreCase("replace") && args.length >= 10) {
+                placementMode = 5;
+            }
+        }
+        if (placementMode == 0 || placementMode == 3 || placementMode == 4 || placementMode == 5) {
+            for (placeX = xmin; placeX <= xmax; placeX++) {
+                for (placeY = ymin; placeY <= ymax; placeY++) {
+                    for (placeZ = zmin; placeZ <= zmax; placeZ++) {
+                        Location location = CommandUtils
+                            .getLocation(CommandUtils.getLocation(sender), Double.toString(placeX), Double.toString(placeY), Double.toString(placeZ));
+                        if (!placeBlock(location, dataValue, args, type, commandMessages, sender, placementMode)) {
+                            return false;
+                        }
+                    }
+                }
+            }
+        }
+        if (placementMode == 1 || placementMode == 2) {
+            List<Location> blocks = new ArrayList<>();
+            for (placeX = xmin; placeX <= xmax; placeX++) {
+                for (placeY = ymin; placeY <= ymax; placeY++) {
+                    for (placeZ = zmin; placeZ <= zmax; placeZ++) {
+                        Location location = CommandUtils
+                            .getLocation(CommandUtils.getLocation(sender), Double.toString(placeX), Double.toString(placeY), Double.toString(placeZ));
+                        blocks.add(location);
+                    }
+                }
+            }
+            xmin += 1;
```

Comment:
Use `++` and `--` instead. Although I'm starting to think this could make use/extend the [`BoundingBox`](https://github.com/GlowstoneMC/Glowstone/blob/dev/src/main/java/net/glowstone/entity/physics/BoundingBox.java) utilities somehow.